### PR TITLE
fix: Wrong error message when inviting

### LINKED ISSUE #15333: [CAL-3863] Failure to invite a user shows wrong error message
User part of other organization error for an email that is already a member of the organization
image

<sub>[CAL-3863](https://linear.app/calcom/issue/CAL-3863/failure-to-invite-a-user-shows-wrong-error-message)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
- Fixes #15333

It was causing a lot of confusion.

[Loom ](https://www.loom.com/share/41febe8b8e4049ab8c92b9c64ca5330e)

## Mandatory Tasks (DO NOT REMOVE)
- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?
See the loom

### CODE REVIEW

**Comment by hariombalhara** [comment]
* **#15332** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/15332?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hariombalhara and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts [hunk 1]
```diff
@@ -19,16 +19,17 @@ import type { TeamWithParent } from "./types";
 import {
   checkPermissions,
   getTeamOrThrow,
-  getUsernameOrEmailsToInvite,
+  getUniqueUsernameOrEmailsOrThrow,
   getOrgConnectionInfo,
-  getIsOrgVerified,
+  getOrgState,
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts:24]: Better names

File: packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts [hunk 2]
```diff
@@ -87,9 +88,16 @@ export const inviteMemberHandler = async ({ ctx, input }: InviteMemberOptions) =
 
   // Existing users have a criteria to be invited
   const existingUsersWithMembershipsThatNeedToBeInvited = existingUsersWithMemberships.filter(
-    (invitee) => invitee.canBeInvited
+    (invitee) => invitee.canBeInvited === INVITE_STATUS.CAN_BE_INVITED
   );
 
+  if (!beSilentAboutErrors && existingUsersWithMemberships[0].canBeInvited !== INVITE_STATUS.CAN_BE_INVITED) {
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/teams/inviteMember/inviteMember.handler.ts:94]: Moved the logic close to the variables being used.

File: packages/features/ee/organizations/pages/settings/other-team-members-view.tsx [hunk 3]
```diff
@@ -215,7 +215,7 @@ const MembersView = () => {
                       if (Array.isArray(data.usernameOrEmail)) {
                         showToast(
                           t("email_invite_team_bulk", {
-                            userCount: data.usernameOrEmail.length,
```
Comment (hariombalhara) [inline packages/features/ee/organizations/pages/settings/other-team-members-view.tsx:218]: `userNameOrEmail` has all the emails that were sent. We need to count only the emails that were invited

**Review by sean-brydon — APPROVED** [review]
Nice one! Tested works fine. Thanks for making this a bit more readable - file is a bit of a mess :D

**Pull request merged**